test/fixtures/api/no-data.json

```diff
@@ -0,0 +1 @@
+{"statusCode":404,"body":"{}"}
```

Comment:
I haven't actually seen the /prod API call fail, so I had to take a guess here as to what it returns if no testing locations have available appointments.

Reply:
Based on https://github.com/livgust/covid-vaccine-scrapers/blob/4210545b850182f573c47fd97248e223ecce5895/main.js#L61-L63

```js
		const responseJson = {
			results: finalResultsArray,
		};
```


and https://github.com/livgust/covid-vaccine-scrapers/blob/4210545b850182f573c47fd97248e223ecce5895/main.js#L80-L88

```js
			const params = {
				Bucket: process.env.AWSS3BUCKETNAME,
				Key: "data.json",
				Body: JSON.stringify(responseJson),
			};

			// Uploading files to the bucket
			return await new Promise((resolve, reject) => {
				s3.upload(params, function (err, data) {
```

at a minimum it'll have inside the body a null `results` array.
I think the better test is to count the length of the results array rather than testing for an explicit value.

Reply:
oh nice, thanks for the links to that repo.  I'll update the fixture to reflect that.

> I think the better test is to count the length of the results array rather than testing for an explicit value.

the length of the results array is just the length of the [mock data used by the unit test](https://github.com/livgust/macovidvaccines.com/pull/1/files#diff-49804f06dcb661d866cd1ae6a0494cb7d99cbe4b2019b149da6a7e0f7b52ccf4R38); testing against that is equivalent to testing that the mock function I wrote is doing the right thing.  the better test would be to count the number of rows rendered by the table but I can't do that since there is currently nothing in the rendered markup for [testing-library](https://testing-library.com/docs/dom-testing-library/api) to hook into (also, no table at all is rendered in the case of API call failure so there's nothing for the test framework to assert against).

I thought about adding those hooks since it would be pretty straightforward and if done right would provide a better experience for more user agents but I was hesitant to parachute into someone else's project and start changing too many things.  (for background, I offered to help with writing tests after a brief discussion with Olivia on Twitter the other day)

Reply:
> the length of the results array is just the length of the mock data used by the unit test; testing against that is equivalent to testing that the mock function I wrote is doing the right thing.

Then I suspect we don't need the test at all.